File: cpp/Instruções Sequenciais/Q04.cpp
/*
------------------------------------------------
4. Crie um algoritmo/fluxograma/programa que lê 
o nome de um aluno, as notas de suas três provas
e calcule e exiba na tela a média harmônica das 
provas.
 */
#include <iostream>
#include <locale.h>
using namespace std;
int main () {
	setlocale(LC_ALL,"Portuguese");
	string nome;
	float nota1, nota2, nota3, media;
	cout<<"Informe o nome do aluno:";
	cin>>nome;
	cout<<"Informe a nota 1:";
	cin>>nota1;
	cout<<"Informe a nota 2:";
	cin>>nota2;
	cout<<"Informe a nota 3:";
	cin>>nota3;	
	media = 3 / (1/nota1 + 1/nota2 + 1/nota3);
	cout<<"A media harmônica de "<<nome<<" é:";
	cout<<media<<endl;
	return 0;
}

File: exercicios/python/Instruções Sequenciais/Q04.py
# Exercício 4 — Média harmônica das três provas
def main():
    nome = input("Informe o nome do aluno: ")
    n1 = float(input("Nota da prova 1: "))
    n2 = float(input("Nota da prova 2: "))
    n3 = float(input("Nota da prova 3: "))
    if n1 == 0 or n2 == 0 or n3 == 0:
        print("Notas não podem ser zero para cálculo da média harmônica.")
    else:
        media_harmonica = 3 / (1/n1 + 1/n2 + 1/n3)
        print(f"Aluno: {nome} — Média harmônica: {media_harmonica}")

if __name__ == "__main__":
    main()